gubernator/pull_request.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def builds_to_table(jobs):
+    """
+    Convert a list of builds into arguments suitable for rendering.
+
+    Args:
+        jobs: a dict of {job: [(build, started.json, finished.json), ...]}
+    Returns:
+        max_builds: the number of build columns
+        headings: a list of [(version, column width, timestamp)]
+        rows: a list of [(build, [(number, status) or None])]
+    """
+
+    def commit(version):
+         return version.split('+')[-1]
+
+    # Compute the headings first -- versions and their maximum build counts.
+
+    versions = {}
```

Comment:
In lieu of types, I would appreciate a more detailed comment saying what these two dicts are.